### DISCUSSION TITLE
Consider deprecating `module_from_template`

### DISCUSSION THREAD

**Discussion by Archmonger** [idom-team/idom#821] [Ideas]
While the ability to load JavaScript modules from NPM on the fly is clever, it technologically isn't possible to do properly.

That is due to ReactJS version mismatches, which frequently pop up when using this API.

For the health of this framework, I would recommend we deprecate `module_from_template` and double-down on _JavaScript Components API_ docs to supplement this.

**Comment by rmorshea** [comment]
I think I agree. An [`idom-build`](https://github.com/idom-team/idom/issues/786) cli will also help with this.
